docs/wpf/getting-started.md

```diff
@@ -1 +1,299 @@
+
+### RegisterTypes
+
+This function is used to register any app dependencies. For example, there might be an interface to read customer data from a persistent store of some kind and the implementation of it is to use a database of some kind. It might look something like this:
+
+```cs
+public interface ICustomerStore
+{
+    List<string> GetAll();
+}
+
+public class DbCustomerStore()
+{
+    public List<string> GetAll()
+    {
+        // return list from db
+    }
+}
+```
+
+Objects in the app, such as view models, that have a dependency on the customer data would require an ```ICustomerStore``` object. In the ```App.RegisterTypes``` function, a registration would be made to create a ```DbCustomerStore``` every time an object takes a dependency on ```ICustomerStore```.
+
+```cs
+protected override void RegisterTypes(IContainerRegistry containerRegistry)
+{
+    containerRegistry.Register<Services.ICustomerStore, Services.DbCustomerStore>();
+    // register other needed services here
+}
+```
+
+> IContainerRegistry has other functions for registering against interfaces as well. ```RegisterInstance``` will register a created instance of an object against an interface. In effect the implementation of the registered interface is a singleton. A similar method is ```RegisterSingleton``` that will create a single instance at the time the dependency is made and not before. It should be noted that the ```Container``` can also resolve concrete types without a prior registration.
+
+### CreateShell
+
+The second method that has to be implemented is the CreateShell method. This is the method that will create the main window of the application. The Container property of the App class should be used to create the window as it takes care of any dependencies.
+
+```cs
+public partial class App : PrismApplication
+{
+    // RegisterTypes function is here
+
+    protected override Window CreateShell()
+    {
+        var w = Container.Resolve<MainWindow>();
+        return w;
+    }
+}
+```
+
+At this point, the app can be built and run and should look like the following:
+
+![First Run of App](images/FirstRun.PNG)
+
+This now a Prism app. There isn't much here yet, but there are lots of things that Prism can help out with, such as breaking up the app into manageable chunks, navigation and implementing the MVVM patterns.
+
+## View Models
+
+WPF is well setup to use an MVVM pattern and Prism helps a lot with this. It has a base class that handles the INotifyPropertyChanged infrastructure that publishes changes from the view model to the view. There are some other classes that make it simple to handle buttons from within the view model as opposed to writing an event handler in your code behind.
+
+First there needs to be some controls added to the view. Go to ```MainWindow.xaml``` and add the following markup inside the ```<Grid></Grid>``` element.
```

Comment:
I would actually provide the full grid snippet and instruct readers to use that as the content of the MainWindow

Reply:
Updated this.